### ISSUE TITLE
Autoinstall tasks

### ISSUE THREAD

**Issue by frontsideair** [sapegin/mrm#97]
Hi,

I've created a task `mrm-task-yarnhook` to make installation of `yarnhook` a one-liner. (Awesome project by the way!) But when I try to see if it works, I'm getting a confusing error message. 

```
Task “yarnhook” not found.

We’ve tried these locations:

- /Users/fatih/dotfiles/mrm/yarnhook/index.js
- /Users/fatih/.mrm/yarnhook/index.js
- “yarnhook” in the default mrm tasks
- npm install -g mrm-task-yarnhook
- npm install -g yarnhook
```

Reading the [sharing tasks](https://mrm.js.org/docs/sharing-tasks) doc page, I would expect `mrm` to find `mrm-task-yarnhook` automatically. It seems like this task must be included in default tasks or must be globally installed, which would make installation more than one line.

I wonder if `mrm` can automatically find the task using `npm`, which would be very useful. Otherwise it would be nice to update the documentation to make this clear. In that case, would it be okay to add `mrm-task-yarnhook` to the default task set?

Best

**Comment by sapegin** [comment]
Looks like we have autoinstall only for presets now: #92. Feel free to send a pull request! 👾

**Comment by frontsideair** [comment]
I'm a bit confused if I should add `mrm-task-yarnhook` to the default preset or if I should add the functionality to globally install any task. I have a branch ready for the former but wanted to clarify before I opened a PR.

**Comment by sapegin** [comment]
The latter: we already have this feature for _presets_ so we need to extend it for _tasks_ too.

**Comment by frontsideair** [comment]
I did a bit of research to find a way to require tasks without installing them globally, which is too intrusive in my opinion. I read through the issues and looks like everyone agrees with that, but it was the only way that works.

I had a thought that we can copy the way `npx` executes packages without installing them. I made a [proof-of-concept](https://gist.github.com/frontsideair/60cb2e51e32d7a6d0d4ec0e26e57e20a) using `libnpx` and seems like it works nicely. Would you like this turned into a PR?

**Comment by sapegin** [comment]
What's wrong with the way we autoinstall presets? Not sure now npx way is better.

**Comment by frontsideair** [comment]
I personally don't like to install packages globally so they don't pollute my system, but for others it might be impossible if they don't have root privileges or they could find it to be a nuisance. Requiring root privileges could be a security risk too.

**Comment by sapegin** [comment]
That makes sense! But in this case, we'll need to use npx for both: tasks and presets.

**Comment by frontsideair** [comment]
Yup, that's also what I had in mind. I'll start this soon as I have your blessing!

**Comment by sapegin** [comment]
Let's try and see how it works ;-)

**Comment by brettz9** [comment]
I might suggest renaming this issue "Autoinstall tasks". And sounds great!

**Comment by frontsideair** [comment]
Fixed by #99
